examples/tutorials/advanced/draping_on_3d_surface.py

```diff
@@ -0,0 +1,177 @@
+# colormap specified via the ``cmap`` parameter applies to the grid passed to
+# ``drapegrid``, not to ``grid``. The azimuth and elevation a the 3-D plot are set via
+# the ``perspective`` parameter.
+
+fig = pygmt.Figure()
+
+# Set up colormap for the crustal age
+pygmt.config(COLOR_NAN="lightgray")
+pygmt.makecpt(cmap="batlow", series=[0, 200, 1], reverse=True, overrule_bg=True)
+
+fig.grdview(
+    projection="M12c",  # Mercator projection with a width of 12 centimeters
+    region=region_3d,
+    grid=grd_relief,  # Use elevation grid for z values
+    drapegrid=grd_age,  # Use crustal age grid for color-coding
+    cmap=True,  # Use colormap created for the crustal age
+    surftype="i",  # Create an image plot
+    # Use an illumination from the azimuthal directions 0° (north) and 270°
+    # (west) with a normalization via a cumulative Laplace distribution for
+    # the shading
+    shading="+a0/270+ne0.6",
+    perspective=[157.5, 30],  # Azimuth and elevation for the 3-D plot
+    zsize="1.5c",
+    plane="+gdarkgray",
+    frame=True,
+)
+
+# Add colorbar for the crustal age
+fig.colorbar(frame=["x+lseafloor crustal age", "y+lMyr"], position="+n")
+
+fig.show()
+
+
+# %%
+# 2. Draping an image
+# -------------------
+#
+# In the second example, the flag of the European Union (EU) is plotted on top of a
+# topographic map of northwest Europe. This example is modified from
+# :gmt-docs:`GMT example 32 </gallery/ex32.html>`.
+
+# Define study area in degrees East or North
+region_2d = [3, 9, 50, 54]  # [lon_min, lon_max, lat_min, lat_max]
+
+# Download elevation grid for the study region with a resolution of 30 arc-seconds and
+# pixel registration and load it into an xarray.DataArray
+grd_relief = pygmt.datasets.load_earth_relief(resolution="30s", region=region_2d)
+
+# Determine the 3-D region from the minimum and maximum values of the relief grid
+region_3d = [*region_2d, grd_relief.min().to_numpy(), grd_relief.max().to_numpy()]
+
+# Download an PNG image of the flag of the EU using rasterio and load it into a
+# xarray.DataArray
+url_to_image = "https://upload.wikimedia.org/wikipedia/commons/thumb/b/b7/Flag_of_Europe.svg/1000px-Flag_of_Europe.svg.png"
+with rasterio.open(url_to_image) as dataset:
+    data = dataset.read()
+    drapegrid = xr.DataArray(data, dims=("band", "y", "x"))
+
+# %%
+# Again we create a 3-D plot with :meth:`pygmt.Figure.grdview` and passe an Earth relief
+# grid to the ``grid`` parameter to create the topographic surface. But now we pass the
+# PNG image which was loaded into an :class:`xarray.DataArray` to the ``drapgrid``
+# parameter.
+
+fig = pygmt.Figure()
+
+# Set up a colormap with two colors for the EU flag: blue (0/51/153) for the background
+# (value 0 in the nedCDF file -> lower half of 0-255 range) and yellow (255/204/0) for
+# the stars (value 255 -> upper half)
+pygmt.makecpt(cmap="0/51/153,255/204/0", series=[0, 256, 128])
+
+fig.grdview(
```

Comment:
Did you forget `zsize` here?

Reply:
I removed the `zsize` parameter (temporarily), as I had not figured out yet, how to adjust the position of the following plotting elements (added via `coast`, `plot`, `text`) corresponding to the value of `zsize`:

| no `zsize` |  `zsize="1c"` |
| --- | --- |
| ![flag_0c](https://github.com/user-attachments/assets/b97f73fd-230c-4a93-a2cb-39a83b4e0aa8) | ![flag_1c](https://github.com/user-attachments/assets/34de769f-5444-4914-b602-04647bf077ab) |

| `zsize="1.5c"` | `zsize="2c"` |
| --- | --- |
| ![flag_1 5c](https://github.com/user-attachments/assets/47e8897f-c078-40b9-928b-fd9ed6266f21) | ![flag_2c](https://github.com/user-attachments/assets/9be3f18c-7004-4f53-ac74-5bbded0fdef5) |

Reply:
For points, `plot3d` is needed. For coastlines, it's likely we need to dump the coastlines and the use `grdtrack` to determine the z-value for each data points and then pass them to `plot3d`.